core/reflection.jvm/src/kotlin/reflect/jvm/internal/KDeclarationContainerImpl.kt

```diff
@@ -236,7 +236,17 @@ internal abstract class KDeclarationContainerImpl : ClassBasedDeclarationContain
         repeat((valueParameters.size + Integer.SIZE - 1) / Integer.SIZE) {
             result.add(Integer.TYPE)
         }
-        result.add(if (isConstructor) DEFAULT_CONSTRUCTOR_MARKER else Any::class.java)
+
+        if (isConstructor) {
+            // Constructors that include the value class as an argument will include DEFAULT_CONSTRUCTOR_MARKER as an argument,
+            // regardless of whether there is a default argument.
+            // On the other hand, when searching for the default constructor,
+            // DEFAULT_CONSTRUCTOR_MARKER needs to be present only at the end, so it is removed here.
+            // Note here that DEFAULT_CONSTRUCTOR_MARKER is counted as one argument in the current bytecode generation,
+            // so it must not be removed before calculating the size of the masks.
```

Comment:
I am very sorry, but I am not fluent in English.
If there is a better way to express this, I would appreciate it if you could let me know.

Reply:
> // Note here that DEFAULT_CONSTRUCTOR_MARKER is counted as one argument in the current bytecode generation,
> // so it must not be removed before calculating the size of the masks.

I deleted the above because it was my misunderstanding.
[58eb137](https://github.com/JetBrains/kotlin/pull/4746/commits/58eb13794d4fcc8e72abec449c6aaa555da99674)